allennlp/tools/drop_eval.py

```diff
@@ -145,11 +152,11 @@ def answer_json_to_strings(answer: Dict[str, Any]) -> Tuple[Tuple[str, ...], str
     Takes an answer JSON blob from the DROP data release and converts it into strings used for
     evaluation.
     """
-    if answer["number"]:
+    if "number" in answer and answer["number"]:
         return tuple([str(answer["number"])]), "number"
-    elif answer["spans"]:
+    elif "spans" in answer and answer["spans"]:
         return tuple(answer["spans"]), "span" if len(answer["spans"]) == 1 else "spans"
-    else:
+    elif "date" in answer:
         return tuple(["{0} {1} {2}".format(answer["date"]["day"],
                                            answer["date"]["month"],
                                            answer["date"]["year"])]), "date"
```

Comment:
Mypy is complaining because you don't have an `else` in here that either returns `None` or raises an error.